src/lib/statistics/chunk_statistics/histograms/abstract_histogram.hpp

```diff
@@ -0,0 +1,203 @@
+
+  virtual HistogramType histogram_type() const = 0;
+
+  /**
+   * Returns a string with detailed information about the histogram, including the edges of the individual bins.
+   */
+  std::string description() const;
+
+  /**
+   * Returns the estimated selectivity given a predicate type and its parameter(s).
+   * It will always be between 0 and 1.
+   */
+  float estimate_selectivity(const PredicateCondition predicate_type, const AllTypeVariant& variant_value,
+                             const std::optional<AllTypeVariant>& variant_value2 = std::nullopt) const;
+
+  /**
+   * Returns the estimated cardinality given a predicate type and its parameter(s).
+   * It will always be between 0 and total_count().
+   * This method is specialized for strings to handle predicates uniquely applicable to string columns.
+   */
+  float estimate_cardinality(const PredicateCondition predicate_type, const AllTypeVariant& variant_value,
+                             const std::optional<AllTypeVariant>& variant_value2 = std::nullopt) const;
+
+  /**
+   * Returns whether a given predicate type and its parameter(s) can be pruned.
+   * This method is specialized for strings to handle predicates uniquely applicable to string columns.
+   */
+  bool can_prune(const PredicateCondition predicate_type, const AllTypeVariant& variant_value,
+                 const std::optional<AllTypeVariant>& variant_value2 = std::nullopt) const override;
+
+  /**
+   * Returns the lower bound (minimum value) of the histogram.
+   * This is equal to the smallest value in the segment.
+   */
+  T min() const;
+
+  /**
+   * Returns the upper bound (maximum value) of the histogram.
+   * This is equal to the largest value in the segment.
+   */
+  T max() const;
+
+  /**
+   * Given a value return the next representable value.
+   * This method is a wrapper for the functions in histogram_utils.
+   */
+  T get_next_value(const T value) const;
+
+  /**
+   * Returns the number of bins actually present in the histogram.
+   * This number can differ from the number of bins requested when creating a histogram.
+   */
+  virtual size_t num_bins() const = 0;
+
+  /**
+   * Returns the number of values represented in the histogram.
+   * This is equal to the length of the segment during creation, without null values.
+   */
+  virtual uint64_t total_count() const = 0;
+
+  /**
+   * Returns the number of distinct values represented in the histogram.
+   * This is equal to the number of distinct values in the segment during creation.
+   */
+  virtual uint64_t total_count_distinct() const = 0;
+
+ protected:
+  /**
+   * Returns a list of pairs of distinct values and their respective number of occurrences in a given segment.
+   * The list is sorted by distinct value from lowest to highest.
+   */
+  static std::vector<std::pair<T, uint64_t>> _calculate_value_counts(const std::shared_ptr<const BaseSegment>& segment);
```

Comment:
function name is pretty unspeaking. "_value_counts_in_segment" or similar, maybe?